File: Data Structures & Algorithms/reverse-string/submission-2.java
class Solution {
    public void reverseString(char[] s) {
        recursiveReverse(s, 0, s.length - 1);
    }

    public void recursiveReverse(char[] s, int left, int right) {
        if (left >= right) return;
        
        final char stored = s[left];
        
        s[left] = s[right];
        s[right] = stored;

        recursiveReverse(s, left + 1, right - 1);
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = 0
        
        maximum = 0
        counts = {}
        maximumFrequency = 0
        for right in range(len(s)):
            counts[s[right]] = 1 + counts.get(s[right], 0)
            maximumFrequency = max(maximumFrequency, counts[s[right]])

            while (right - left + 1) - maximumFrequency > k:
                counts[s[left]] -= 1
                left += 1
            maximum = max(maximum, right - left + 1)
        return maximum